### PR TITLE
 Include component stack in more places, including SSR

### PR DESCRIPTION
**Pull request by gaearon**
This includes component stack instead of owner name in more places, including previously TODO'd places in the server renderer. It also gets rid of the last SSR dependency on reconciler modules so I could remove `react-reconciler/*` from the source whitelist for SSR bundles.

### CODE REVIEW

File: packages/react-dom/src/server/ReactPartialRenderer.js [hunk 1]
```diff
@@ -80,9 +82,9 @@ if (__DEV__) {
     currentDebugStack = null;
     ReactDebugCurrentFrame.getCurrentStack = null;
   };
-  var getStackAddendum = function(): null | string {
+  getStackAddendum = function(): null | string {
     if (currentDebugStack === null) {
-      return null;
+      return '';
```
Comment (bvaughn) [inline packages/react-dom/src/server/ReactPartialRenderer.js]: I'm curious: Why the change from `null` => `''`?

(Is it to maintain the previous behavior where we had an inline function just returning an empty string?)

File: packages/react-dom/src/shared/__tests__/ReactDOMComponent-test.js [hunk 2]
```diff
@@ -954,50 +954,38 @@ describe('ReactDOMComponent', () => {
...
           <input dangerouslySetInnerHTML={{__html: 'content'}} />,
           container,
         );
-      }).toThrowError(
-        'input is a void element tag and must neither have `children` nor use ' +
-          '`dangerouslySetInnerHTML`.',
-      );
-    });
-
-    it('should include owner rather than parent in warnings', () => {
```
Comment (gaearon) [inline packages/react-dom/src/shared/__tests__/ReactDOMComponent-test.js]: Removed this test since it's not really useful in component stacks (where we have both).
Would be nice to mark owner in some special way in stacks but that's a separate issue.

File: packages/react-dom/src/shared/__tests__/ReactDOMComponent-test.js [hunk 3]
```diff
@@ -1178,19 +1166,27 @@ describe('ReactDOMComponent', () => {
       expect(tracker.getValue()).toEqual('foo');
     });
 
-    it('should warn for children on void elements', () => {
+    it('should throw for children on void elements', () => {
```
Comment (gaearon) [inline packages/react-dom/src/shared/__tests__/ReactDOMComponent-test.js]: Just describing what it really tested for.

File: packages/react-dom/src/shared/__tests__/ReactServerRendering-test.js [hunk 4]
```diff
@@ -724,4 +734,16 @@ describe('ReactDOMServer', () => {
         'HTML tags in React.',
     );
   });
+
+  it('should warn about contentEditable and children', () => {
```
Comment (gaearon) [inline packages/react-dom/src/shared/__tests__/ReactServerRendering-test.js]: Not new behavior, but we didn't have an SSR test for it.

File: packages/react-dom/src/shared/__tests__/ReactDOMComponent-test.js (hunk 3 above)
Comment (bvaughn) [inline packages/react-dom/src/shared/__tests__/ReactDOMComponent-test.js]: 👍

File: packages/react-dom/src/server/ReactPartialRenderer.js (hunk 1 above)
Comment (gaearon) [reply packages/react-dom/src/server/ReactPartialRenderer.js]: Because we use `getStack()` directly as `%s`, which would show up as `null` right in the message when missing. I agree it's not great though and would be nice to fix the whole thing by adding a separate `warningWithStack` or something.

Comment (bvaughn) [reply packages/react-dom/src/server/ReactPartialRenderer.js]: Ah. Gotcha.

I asked b'c the Flow return type is also `null | string` but it's no big deal 😄

File: packages/react-dom/src/shared/utils/assertValidProps.js [hunk 5]
```diff
@@ -18,11 +18,7 @@ if (__DEV__) {
 
 var HTML = '__html';
 
-function assertValidProps(
-  tag: string,
-  props: ?Object,
-  getStack: () => string,
-) {
+function assertValidProps(tag: string, props: ?Object, getStack: () => string) {
```
Comment (bvaughn) [inline packages/react-dom/src/shared/utils/assertValidProps.js]: 😆  "gosh"
